stellar-contract-env-common/src/raw_val.rs

```diff
@@ -42,7 +42,7 @@ pub enum Tag {
 }
 
 #[repr(transparent)]
-#[derive(Copy, Clone)]
+#[derive(Copy, Clone, PartialEq, Eq)]
```

Comment:
Can you add a comment pointing out that RawVal equality if based on the RawVal alone and not the host, so two equal objects may be referenced by inequal RawVals. I think this behavior is appropriate, it's too inconvenient to never be able to compare them and we shouldn't need a host to do comparisons.

Reply:
@leighmcculloch I added the comment.

Reply:
I agree that `obj_cmp` is the proper way to compare objects, but people will inadvertently use this by accident and then there will be bugs. I'm really not sure I'm comfortable with this change.

Reply:
Removing the Eq from ConversionError will make it harder to write tests for any function that can return the error.

Reply:
@graydon A while ago you mentioned the possibility for interning host object types. Is that something planned? If host objects are interned then this problem would no longer be relevant.